src/java.base/unix/native/libjli/java_md_common.c

```diff
@@ -128,8 +128,12 @@ Resolve(char *indir, char *cmd)
 {
     char name[PATH_MAX + 2], *real;
 
-    if ((JLI_StrLen(indir) + JLI_StrLen(cmd) + 1)  > PATH_MAX) return 0;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+    if ((JLI_StrLen(indir) + JLI_StrLen(cmd) + 2) > sizeof(name)) return 0;
     JLI_Snprintf(name, sizeof(name), "%s%c%s", indir, FILE_SEPARATOR, cmd);
+#pragma GCC diagnostic pop
```

Comment:
Can we just replace this code rather than putting pragmas here?

Reply:
I tried several patterns, but I couldn't find out a solution other than pragmas. Do you have any ideas?

For example:
```c
    if ((JLI_StrLen(indir) + JLI_StrLen(cmd) + 2) < sizeof(name)) {
      JLI_Snprintf(name, sizeof(name), "%s%c%s", indir, FILE_SEPARATOR, cmd);
    } else {
      return 0;
    }
```

Compiler warnings:
```
/home/ysuenaga/github-forked/jdk/src/java.base/unix/native/libjli/java_md_common.c:135:45: error: '%s' directive output may be truncated writing up to 4094 bytes into a region of size between 2 and 4096 [-Werror=format-truncation=]
  135 |       JLI_Snprintf(name, sizeof(name), "%s%c%s", indir, FILE_SEPARATOR, cmd);
      |                                             ^~
In file included from /usr/include/stdio.h:894,
                 from /home/ysuenaga/github-forked/jdk/src/java.base/share/native/libjli/java.h:29,
                 from /home/ysuenaga/github-forked/jdk/src/java.base/unix/native/libjli/java_md_common.c:26:
In function 'snprintf',
    inlined from 'Resolve' at /home/ysuenaga/github-forked/jdk/src/java.base/unix/native/libjli/java_md_common.c:135:7:
/usr/include/bits/stdio2.h:71:10: note: '__builtin___snprintf_chk' output between 2 and 8190 bytes into a destination of size 4097
   71 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   72 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   73 |                                    __va_arg_pack ());
```